frappe/core/doctype/user/user.py

```diff
@@ -811,6 +813,13 @@ def check_roles_added(self):
 			},
 		)
 
+	def validate_ip_addr(self):
+		# remove whitespace
+		if self.restrict_ip.find(",") != -1:
+			self.restrict_ip = self.restrict_ip.strip()
+		else:
+			self.restrict_ip = ",".join(self.get_restricted_ip_list())
```

Comment:
This can be still simplified. There's no need to check if `","` exists in the string or not. 

There's a reason why most str methods offer sane default behaviour. If `","` doesn't exist in string while splitting, they won't be split. Join also works _sensibly_.

Reply:
split and join are supposed to be dual of each other, so any form of `str.join(str.split(..))`  will just work without any special handling.